sdk/core/core-lro/src/lroPoller/bodyPolling.ts

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { failureStates, LROStatus, RawResponse, successStates } from "./models";
+
+function getProvisioningState(rawResponse: RawResponse): string {
+  const { properties, provisioningState } = rawResponse.body ?? {};
+  const state: string | undefined = properties?.provisioningState ?? provisioningState;
+  return state?.toLowerCase() ?? "succeeded";
+}
+
+export function isBodyPollingDone(rawResponse: RawResponse): boolean {
+  const state = getProvisioningState(rawResponse);
+  if (failureStates.includes(state)) {
+    throw new Error(`Provisioning state: ${state}`);
```

Comment:
can we put some additional details, since I'm not sure it would be obvious to the consumer what to do with an error like this

Reply:
I just added the prefix: `The long-running operation has failed.`. Let me know if you have another suggestion.